File: src/hackerrank/Task3.java
package hackerrank;

import java.util.List;

public class Task3 {

  static int count_fruits(int l, int r, int c, List<Integer> fs) {
    int count = 0;
    for (int f: fs) {
      int pos = c + f;
      if (pos >= l && pos <= r) count++;
    }
    return count;
  }

  public static void countApplesAndOranges(
      int left,
      int right,
      int apple_center,
      int orange_center,
      List<Integer> apples,
      List<Integer> oranges
  ) {
    int apple_count =
        count_fruits(left, right, apple_center, apples);

    int orange_count =
        count_fruits(left, right, orange_center, oranges);

    String s = String.format("%d\n%d\n", apple_count, orange_count);
    //                        ^    ^          |            |
    //                        -----|-----------            |
    //                             |                       |
    //                             \-----------------------/

    System.out.print(s);
  }

}

File: src/hackerrank/Task3.cs
namespace Hackerrank1.hackerrank
{
    public static class Solution
    {
        public static void CountApplesAndOranges(int s, int t, int a, int b, int[] apples, int[] oranges)
        {
            int appleCount = 0;
            int orangeCount = 0;

            foreach (int apple in apples)
            {
                int position = a + apple;
                if (position >= s && position <= t)
                {
                    appleCount = appleCount + 1;
                }
            }
            foreach (int orange in oranges)
            {
                int position = b + orange;
                if (position >= s && position <= t)
                {
                    orangeCount = orangeCount + 1;
                }
            }
            Console.WriteLine(appleCount);
            Console.WriteLine(orangeCount);
        }
    }
}
